docs/gcp_setup.md

```diff
@@ -0,0 +1,287 @@
+```
+SUBNET=<your-subnet>
+CLUSTER_NAME=<cluster-name>
+DP_SVC_ACCOUNT=goe-dataproc
+ZONE=<your-zone>
+
+gcloud iam service-accounts create ${DP_SVC_ACCOUNT} \
+--project=${PROJECT} \
+--description="GOE Dataproc service account"
+
+gcloud projects add-iam-policy-binding ${PROJECT} \
+--member=serviceAccount:${DP_SVC_ACCOUNT}@${PROJECT}.iam.gserviceaccount.com \
+--role=roles/dataproc.worker
+
+gcloud compute networks subnets update ${SUBNET} \
+--project=${PROJECT} --region=${REGION} \
+--enable-private-ip-google-access
+
+gcloud dataproc clusters create ${CLUSTER_NAME} \
+--project ${PROJECT} --region ${REGION} --zone ${ZONE} \
+--bucket ${BUCKET} \
+--subnet projects/${PROJECT}/regions/${REGION}/subnetworks/${SUBNET} \
+--no-address \
+--single-node \
+--service-account=${DP_SVC_ACCOUNT}@${PROJECT}.iam.gserviceaccount.com \
+--master-machine-type n2-standard-16 --master-boot-disk-size 1000 \
+--image-version 2.1-debian11
+```
+
+### Roles
+
+#### goe_gcs_role
+
+Note that the role grant is bound to the staging bucket, no project wide access is granted.
+```
+gcloud iam roles create goe_gcs_role --project ${PROJECT} \
+--title="GOE Cloud Storage Access" \
+--description="GOE permissions to access staging Cloud Storage bucket" \
+--permissions=storage.buckets.get,\
+storage.objects.create,storage.objects.delete,\
+storage.objects.get,storage.objects.list \
+--stage=GA
+
+gcloud storage buckets add-iam-policy-binding gs://${BUCKET} \
+--member=serviceAccount:${SVC_ACCOUNT}@${PROJECT}.iam.gserviceaccount.com \
+--role=projects/${PROJECT}/roles/${ROLE}
+```
+
+#### goe_bq_core_role
+Note that this role is granted at the project level.
+```
+gcloud iam roles create goe_bq_core_role --project ${PROJECT} \
+--title="GOE Core BigQuery Access" \
+--description="GOE permissions for core access to BigQuery" \
+--permissions=bigquery.datasets.get,\
+bigquery.jobs.create,\
+bigquery.tables.get,\
+bigquery.tables.list \
+--stage=GA
+
+gcloud projects add-iam-policy-binding ${PROJECT} \
+--member=serviceAccount:${SVC_ACCOUNT}@${PROJECT}.iam.gserviceaccount.com \
+--role=projects/${PROJECT}/roles/goe_bq_core_role
+```
+
+If GOE is permitted to create datasets then add these extra privileges:
+```
+gcloud iam roles update goe_bq_core_role --project ${PROJECT} \
+--add-permissions=bigquery.datasets.create
+```
+
+#### goe_bq_app_role
+Note that the role grant is bound to the target BigQuery dataset, no project wide access is granted. The `bq` utility is used to grant the role because `gcloud` does not support these granular grants.
+
+Also note that the target dataset must be created *before* executing these commands (see [Generating Dataset DDL](#generating-dataset-ddl) for details).
+```
+gcloud iam roles create goe_bq_app_role --project ${PROJECT} \
+--title="GOE Data Update Access" \
+--description="Grants GOE permissions to read and modify data in an application schema" \
```

Comment:
Do we want to say "dataset" instead of "schema"?

Reply:
Fixed